macos/FluentUI/Button.swift

```diff
@@ -0,0 +1,230 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import AppKit
+
+/// Indicates what style our button is drawn as
+@objc public enum ButtonStyle: Int, CaseIterable {
+    case primaryFilled	// Solid fill color
+    case primaryOutline	// Clear fill color, solid outline
+    case borderless		// Clear fill color, clear outline
+}
+
+// MARK: - Button
+
+/// A fluent styled button, with hover effects and a corner radius.
+@objc(MSFButton)
+open class Button: NSButton {
+
+	@objc public init(title: String, style: ButtonStyle = .primaryFilled) {
+		super.init(frame: .zero)
+		self.title = title
+		self.style = style
+		initialize()
+    }
+	
+	@objc public init(image: NSImage, style: ButtonStyle = .primaryFilled) {
+		super.init(frame: .zero)
+		self.style = style
+		initialize()
+		// Image must be set after `wantsLayer = true` so that the image is on the layer and not the underlying NSButtonCell
```

Comment:
Can we guard/fail this in code?

Reply:
What do you mean? Clients should never be able to set an image before `wantsLayer = true`

Reply:
Well if the order is notable enough that it's worth a comment in code, we might as well add verification so someone refactoring it will know if they screwed it up?

Reply:
Added guards to `willSet` of the `title` and `image` property. 
Question: Is there something better we can do than `exit(-1)` if the guard fails? Not sure if we havor throwing an exception, or preconditionFailure, or that's arbitrary